app/javascript/controllers/keywords_controller.js

```diff
@@ -18,36 +18,17 @@ export default class extends Controller {
     this.containerTargets.forEach((container) => container.classList.remove('keywords-container-open') )
   }
 
-  add(event) {
-    const content = this.selectedTemplateTarget.innerHTML
-      .replace(/TEMPLATE_RECORD/g, new Date().valueOf())
-      .replace(/TEMPLATE_LABEL/g, event.target.value)
-
-    this.searchTarget.value = ''
-    this.addItemTarget.insertAdjacentHTML('beforeend', content)
-    // Remove error indications when keyword(s) added
-    this.containerTarget.classList.remove('is-invalid')
-    this.checkForDuplicates()
-  }
-
   checkForDuplicates() {
     if (this.hasDuplicateKeywords()) {
       this.inputTarget.classList.add('is-invalid')
-      // Adding this custom validation to prevent the a deposit from being submitted. See validate-forms.js
+      // Adding this custom validation to prevent the deposit from being submitted. See validate-forms.js
       this.inputTarget.setCustomValidity('has a duplicate')
     } else {
       this.inputTarget.classList.remove('is-invalid')
       this.inputTarget.setCustomValidity('')
     }
   }
 
-  removeAssociation(event) {
-    const item = event.target.closest(".selection-choice")
-    item.querySelector("input[name*='_destroy']").value = 1
-    item.style.display = 'none'
-    this.checkForDuplicates() // clear out the invalid selection class if there are no duplicates anymore
-  }
-
   hasDuplicateKeywords() {
```

Comment:
this methods isn't called anywhere from any components in use as far as I can tell, instead we will just call `checkForDuplicates` directly from the existing component in use